libs/vm_modules/tests/unit/state_tests.cpp

```diff
@@ -252,6 +236,103 @@ TEST_F(StateTests, test_serialisation_of_complex_type)
   EXPECT_EQ(std::string{"ccc"}, retval->PopFrontOne().Get<Ptr<String>>()->str);
 }
 
+template <typename T>
+std::enable_if_t<!IsPtr<T>::value> ArrayFromVariant(Variant const &array, int32_t expected_size,
```

Comment:
I don't think this overload would compile if it was ever instantiated - gtest ASSERT* macros contain a return statement which yields IIRC a testing::AssertionResult, but SFINAE tries to make this a void function. May be worth writing a test that exercises a nested array of non-pointers.

Reply:
`ASSERT_XYZ(...)` macros can be used in any function/method with return type `void` . The proof  that it would compile just fine is the 2nd specialisation of the `ArrayFromVariant<...>` funtion for `Ptr<IArray>` type right bellow (it also returns `void`), or multiple other test-helper methods/functions where ASSER* marcore are used, like for example `SmartContractTests::CreateContract(...)`.